src/client/App.jsx

```diff
@@ -1,30 +1,92 @@
-import { useEffect, useState } from 'react';
-import './App.css';
-import MovieForm from './components/MovieForm';
-import UserForm from './components/UserForm';
+import { useEffect, useState } from "react";
+import "./App.css";
+import MovieForm from "./components/MovieForm";
+import UserForm from "./components/UserForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
 
 function App() {
   const [movies, setMovies] = useState([]);
 
   useEffect(() => {
     fetch(`${apiUrl}/movie`)
-      .then(res => res.json())
-      .then(res => setMovies(res.data));
+      .then((res) => res.json())
+      .then((res) => setMovies(res.data));
   }, []);
 
+  // REGISTER
   const handleRegister = async ({ username, password }) => {
-    
+    try {
+      const response = await fetch(`${apiUrl}/user/register`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          username,
+          password,
+        }),
+      });
+
+      // response from the server containing id and username
+      const data = await response.json();
+      console.log(data);
+    } catch (error) {
+      console.log("Error");
+    }
   };
 
+  // LOGIN
   const handleLogin = async ({ username, password }) => {
-    
+    try {
+      const response = await fetch(`${apiUrl}/user/login`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          username,
+          password,
+        }),
+      });
+
+      const data = await response.json();
+      localStorage.setItem("accessToken", data.token);
+      console.log("token", data.token);
+    } catch (error) {
+      console.log("Error");
+    }
   };
-  
+
+  // MOVIE
   const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
-  }
+    const token = localStorage.getItem("accessToken");
+
+    try {
+      const response = await fetch(`${apiUrl}/movie`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
+        body: JSON.stringify({
+          title,
+          description,
+          runtimeMins,
+        }),
+      });
+
+      const movie = await response.json();
+      console.log("movie", movie);
+
+      // Fetch the updated movie list after adding a movie
+      fetch(`${apiUrl}/movie`)
+        .then((res) => res.json())
+        .then((res) => setMovies(res.data));
+    } catch (error) {
+      console.log("Error");
+    }
+  };
 
```

Comment:
We handled our functions very similarly! Good job, I think it looks very tidy and clean! Also great job on the comments!